docker/Dockerfile

```diff
@@ -1,55 +1,81 @@
+RUN apt-get -y install build-essential \
+  && apt-get -y install cmake curl git file gfortran-12 ksh m4 python3 tcsh time wget vim \
+  && apt-get -y install libnetcdf-pnetcdf-19 libnetcdff7 libnetcdf-dev libnetcdff-dev libxml2 \
+  && apt-get -y install python3-pip python3.11-venv
 
-# Obtain CCPP SCM source code
-RUN cd /comsoftware \
-  && git clone --recursive -b release/public-v6 https://github.com/NCAR/ccpp-scm
-
-# Obtain static data that was previously stored in repository
-RUN cd /comsoftware/ccpp-scm/ \
-  && . contrib/get_all_static_data.sh
-
-# Obtain the pre-computed look-up tables for running with Thompson microphysics
-RUN cd /comsoftware/ccpp-scm/ \
-  && . contrib/get_thompson_tables.sh
-  
-# Run the machine setup script to set environment variables
-ENV CC=/opt/rh/devtoolset-9/root/usr/bin/gcc
-ENV CXX=/opt/rh/devtoolset-9/root/usr/bin/g++
-ENV F77=/opt/rh/devtoolset-9/root/usr/bin/gfortran
-ENV F90=/opt/rh/devtoolset-9/root/usr/bin/gfortran
-ENV FC=/opt/rh/devtoolset-9/root/usr/bin/gfortran
-
-ENV NETCDF=/comsoftware/libs/netcdf
-
-RUN cd /comsoftware/ccpp-scm/contrib \
-  && wget https://raw.githubusercontent.com/NCAR/ccpp-scm/3f501aa8af0fb00ff124d8301c932292d1d0abf3/contrib/build_nceplibs.sh \
-  && chmod +x build_nceplibs.sh \
-  && cd .. \
-  && ./contrib/build_nceplibs.sh $PWD/nceplibs
-
-ENV bacio_ROOT /comsoftware/ccpp-scm/nceplibs
-ENV sp_ROOT /comsoftware/ccpp-scm/nceplibs
-ENV w3nco_ROOT /comsoftware/ccpp-scm/nceplibs
-
-# Create your own link from python -> python3 
-# This works without setting the system PATH env var
-# since /usr/local/bin is before /usr/bin in the search path.
-USER root
-RUN ln -s /usr/bin/python3 /usr/local/bin/python
+MAINTAINER Grant Firl <grantf@ucar.edu> or Michael Kavulich <kavulich@ucar.edu>
+
+#Compiler environment variables
+ENV CC  /usr/bin/gcc
+ENV FC  /usr/bin/gfortran
+ENV CXX /usr/bin/g++
+ENV F77 /usr/bin/gfortran
+ENV F90 /usr/bin/gfortran
+
+# Other necessary environment variables
+ENV LD_LIBRARY_PATH /usr/lib/
+
+# Set up unpriviledged user account, set up user home space and make sure user has permissions on all stuff in /comsoftware
+RUN groupadd comusers -g 9999 \
+ && useradd -u 9999 -g comusers -M -s /bin/bash -c "Unpriviledged user account" -d /home comuser \
+ && mkdir /comsoftware \
+ && chown -R comuser:comusers /home \
+ && chmod 6755 /home \
+ && chown -R comuser:comusers /comsoftware \
+ && chmod -R 6755 /comsoftware
+
+# Link version-specific aliases (python3 will be created later with virtual environment)
+RUN ln -s ~comuser/.venv/bin/python3 /usr/local/bin/python
+RUN ln -s /usr/bin/gfortran-12 /usr/bin/gfortran
+
+# all root steps completed above, now continue below as regular userID comuser
 USER comuser
+WORKDIR /home
+
+# Build NCEP libraries we need for SCM
+
+ENV NCEPLIBS_DIR /comsoftware/nceplibs
 
-# Invoke cmake on the source code to build
-RUN cd /comsoftware/ccpp-scm/scm \
+RUN mkdir -p $NCEPLIBS_DIR/src && cd $NCEPLIBS_DIR/src \
```

Comment:
Why are we not trying to use spack-stack?

Reply:
Spack-stack is a bit bloated for a container since it has a lot of software we don't use, especially since @scrasmussen is removing these NCEPLIBS dependencies soon.